main.py

```diff
@@ -0,0 +1,178 @@
+"""
+ Shrinks a Linux file system, eg. ext4. Note that XFS is not supported.
+"""
+
+import os
+import sys
+import subprocess
+import argparse
+from collections import namedtuple
+
+
+SHRINK_TAG = 'x-systemd.shrinkfs'
+
+fstab_entry = namedtuple("fstab_entry",
+                         "device mount fs_type options dump passno")
+
+
+def init_arguments():
+    """Init arguments"""
+    parser = argparse.ArgumentParser(description='Shrinks an unmounted' +
+                                     ' partition to a given size defined' +
+                                     ' with option ' + SHRINK_TAG +
+                                     ' in the /etc/fstab entry of the device')
+    parser.add_argument('device_name', type=str,
+                        help='name of the device as found in the first' +
+                        ' column in the /etc/fstab')
+    return parser
+
+
+def main():
+    """Main funtion"""
+    # init arguments
+    parser = init_arguments()
+    # process arguments
+    args = parser.parse_args()
+    if len(args.device_name) == 0:
+        print("Missing device name argument")
+        sys.exit(1)
+    with open("/etc/fstab", "r") as file:
+        entries = read_fstab(file.readlines())
+    found = False
+    for entry in entries:
+        if entry.device == args.device_name:
```

Comment:
Can this happen more than once? If yes, is this a problem? If not, then you can just return inside the `if` after `process_entry` and then you don't need to maintain `found`

Reply:
Found is used to inform the user in case the device was not found in the fstab. I'll keep `found` but I will break from the loop if the device is found instead of continuously looping.

Reply:
No need to maintain `found` if you return from inside the `if` because then if you got out of the loop, it means that the device was not found

Reply:
sounds good